consensus/spos/worker.go

```diff
@@ -748,6 +785,31 @@ func (wrk *Worker) checkChannels(ctx context.Context) {
 				}
 			}
 		}
+
+		wrk.callReceivedHeaderCallbacks(rcvDta)
+	}
+}
+
+func (wrk *Worker) callReceivedHeaderCallbacks(message *consensus.Message) {
+	headerMessageType := wrk.consensusService.GetMessageTypeBlockHeader()
+	if message.MsgType != int64(headerMessageType) || !wrk.enableEpochsHandler.IsFlagEnabled(common.EquivalentMessagesFlag) {
```

Comment:
maybe check IsFlagEnabledInEpoch, after header is decoded?

Reply:
IsFlagEnabled should be faster then decoding the header, this is why I did that first